zipline/pipeline/loaders/blaze/earnings.py

```diff
@@ -1,29 +1,14 @@
-from datashape import istabular
-import pandas as pd
-from toolz import valmap
-
 from .core import (
+    ANNOUNCEMENT_FIELD_NAME,
     TS_FIELD_NAME,
-    SID_FIELD_NAME,
-    bind_expression_to_resources,
-    ffill_query_in_range,
+    SID_FIELD_NAME
 )
 from zipline.pipeline.data import EarningsCalendar
-from zipline.pipeline.loaders.base import PipelineLoader
 from zipline.pipeline.loaders.earnings import EarningsCalendarLoader
-from zipline.pipeline.loaders.utils import (
-    check_data_query_args,
-    normalize_data_query_bounds,
-    normalize_timestamp_to_query_time,
-)
-from zipline.utils.input_validation import ensure_timezone, optionally
-from zipline.utils.preprocess import preprocess
-
+from .events import BlazeEventsLoader
 
-ANNOUNCEMENT_FIELD_NAME = 'announcement_date'
 
-
-class BlazeEarningsCalendarLoader(PipelineLoader):
+class BlazeEarningsLoader(BlazeEventsLoader):
```

Comment:
Because the concrete loader is called `EventsLoader`, so it seems like these 2 loaders should follow a pattern.